cookbooks/aws-parallelcluster-install/recipes/c_states.rb

```diff
@@ -0,0 +1,83 @@
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: c_states
+#
+# Copyright:: 2013-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+return unless node['kernel']['machine'] == 'x86_64'
+
+ruby_block 'Append GRUB_CMDLINE_LINUX_DEFAULT if it do not exist in /etc/default/grub' do
+  block do
+    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
+    command = 'grep "GRUB_CMDLINE_LINUX_DEFAULT=" /etc/default/grub'
+    command_out = shell_out(command).stdout
+
+    if command_out.include? "GRUB_CMDLINE_LINUX_DEFAULT="
+      Chef::Log.info('Found GRUB_CMDLINE_LINUX_DEFAULT line')
+    else
+      Chef::Log.warn('GRUB_CMDLINE_LINUX_DEFAULT not found - Adding')
+      command = 'echo \'GRUB_CMDLINE_LINUX_DEFAULT=""\' >> /etc/default/grub'
+      shell_out(command)
+      Chef::Log.info('Added GRUB_CMDLINE_LINUX_DEFAULT line')
+    end
+  end
+  action :run
+end
+
+ruby_block 'Add "intel_idle.max_cstate=1" to /etc/default/grub in line GRUB_CMDLINE_LINUX_DEFAULT' do
+  block do
+    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
+    command = 'grep "GRUB_CMDLINE_LINUX_DEFAULT=" /etc/default/grub'
+    command_out = shell_out(command).stdout
+
+    if command_out.include? "intel_idle.max_cstate=1"
+      Chef::Log.info('Found intel_idle.max_cstate=1 in GRUB_CMDLINE_LINUX_DEFAULT')
+    else
+      Chef::Log.warn('intel_idle.max_cstate=1 not found - Adding')
+      command = 'sed -i \'s/^\(GRUB_CMDLINE_LINUX_DEFAULT=".*\)"$/\1 intel_idle.max_cstate=1"/g\' /etc/default/grub'
+      shell_out(command)
+      Chef::Log.info('Added intel_idle.max_cstate=1 to GRUB_CMDLINE_LINUX_DEFAULT')
+    end
+  end
+  action :run
+end
+
+ruby_block 'Add "processor.max_cstate=1" to /etc/default/grub in line GRUB_CMDLINE_LINUX_DEFAULT' do
+  block do
+    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
+    command = 'grep "GRUB_CMDLINE_LINUX_DEFAULT=" /etc/default/grub'
+    command_out = shell_out(command).stdout
+
+    if command_out.include? "processor.max_cstate=1"
+      Chef::Log.info('Found processor.max_cstate=1 in GRUB_CMDLINE_LINUX_DEFAULT')
+    else
+      Chef::Log.warn('processor.max_cstate=1 not found - Adding')
+      command = 'sed -i \'s/^\(GRUB_CMDLINE_LINUX_DEFAULT=".*\)"$/\1 processor.max_cstate=1"/g\' /etc/default/grub'
+      shell_out(command)
+      Chef::Log.info('Added processor.max_cstate=1 to GRUB_CMDLINE_LINUX_DEFAULT')
+    end
+  end
+  action :run
+end
+
+case node['platform']
```

Comment:
nit - you can first define the command for platform using value_for_platform
e.g.
```
regen_grub_command = value_for_platform(
  'ubuntu' => {
    'default' => '...',
  },
 ...
)

and then have just one execute using the command variable you have defined

Reply:
Done thank you